README.TXT
=========================================================
#  PHP Navigator 4.12.12
#  Modified: 5:59 PM; December 03, 2008
#            Paul Wratt,
#            Homeless, Melbourne, Austarlia.
#  Created:  5:00 PM; July 25, 2007
#  http://www.isource.net.nz/download.php?phpnav412.zip
#   PHP Navigator Plugins » http://phpnav.isource.net.nz/
#    FileWalker Suite » http://filewalker.isource.net.nz/
#     Original » http://navphp.sourceforge.net/
=========================================================

LATEST: v4.12 (rev 12) now comes in two flavours, FULL and LITE. This reflects the fact that 
        there are now four 'explorers', three editors (four including the work done on 'Atari'), 
        and five 'skin sets', totaling almost 2Mb, making it considerably larger than the original 
        83Kb, and even thats when they are compressed. Part of the reason I started working with
        phpNavigator was because of its small size. Some of the features of the new explorer
        'skins' may be added into the mainstream v4.32, and I will endeavor to at least get 
        the main functions patched to allow 'single page' inserts from this v4.12.

PREVIOUS: v4.12 (rev 8) will now continue life as the prototype bed for v5 development which 
        builds on the v4.2 vision of multi-user and includes advanced skinning and
        supports modules and virtual drives amongst other things. It is also updated to
        be more Linux friendly. There is currently no longer a v4.5 in development.
        v4.15 will become available for historical reasons, once v5 is complete.

UPDATE: This will be the last version of 4.12 (rev 7), with all additions added to 
        PHP Navigator 4.2, a code re-write is required to maintain both a single user
        and multi-user version, within the same codebase, and to allow easier development 
        of further code branches (PHP Media Navigator), keep and eye out for v4.15..
        (v4.2 will also move to v4.5 in the same re-write, then to v5.0 new skinsets)

UPDATE: PHP Navigator 4.2 has been release, and although it includes some of the
        changes I have made, and now includes a multi-user option, I will continue
        to maintain this official v4.1 (previously unofficial 4.2 beta). Also anyone 
        woundering, I wrote a file browser like this in ASP in 1998, and was looking 
        to pick up some PHP skills. Check out my PHP Navigator "tribute" site to 
        see how far I have "gone" from the original 3.21 source code, and my
        FileWalker site for where I am going..

 » User friendly source editors.
 » multiple interface types, including the retro 'Atari'.
 » multiple icon & skin sets supported.
 » single user only.
 » this version a v5 proto-type and hence is unsupported

EDITORS: are "\" fixed. The new proto-type editor 'DevEdit' is a 'code insertion' editor 
      and a work in progress (unsupported). Its intention is to be an Intergrated Development 
      Environment or IDE, with variable 'plain text' fonts, a color coded editor supporting 
      multiple languages, a WYSIWYG editor for point an click design, but mainly a flexible 
      point and click _code_ editor with libraries, eventually supporting at least 9 languages 
      and multiple language API and code libraries. Contributor are welcome, have a look at 
      the source, its pretty straight forward at the moment, 'libraries.???' will eventually use 
      separate includes, and also support non-english interface text.

NOTE: FileWalker is a heavily modified version of PHP Navigator with the read only
      version being code safe against hacking by PHP Navigator users, as well as 
      providing hidden file and folder parsing, although apart from that it is 
      completely PHP Navigator functional, which makes it perfect for either 
      safe public demos of PHP Navigator, or for web hosting services that DO NOT 
      allow file managers.

      It is however designed to sit around (inside the same folder as) PHP Navigator, 
      as are the other versions of FileWalker. PHP Navigator 4.12 is modified with 
      this in mind and will throw you to roFileWalker if you fail a login (if it is 
      installed). It will also show the following modules if they are installed:
        roFileWalker
        Administrator
        Control Panel
        Database
        Archiver
        Imageview
        Mediaman
        & Console

      These, along with SKINS and a few "other" features, is what FileWalker is 
      all about. But I feel it is also necessary to cater for PHP Navigator, since 
      I started playing with 3.21 by adding the dynamic source editor. I hope to 
      see roFileWalker (or a PHP Navigator version) available on the PHP Navigator 
      website, as I know it will be at least as in demand as PHP Navigator 4.12 is.

      PHP Navigator 4.12 will be the only version maintained by me, however I hope to 
      supply Cyril and Timothy with updates that will reflect the quality changes that 
      come about in FileWalker development, especially skins, left panel modules, 
      control panel modules, installer, as well as other plug-in modules along with my 
      other hard core non-Ajax coding (to make it function the same as the Ajax version)
      as I think it is better to compliment than to replace, which is what tends to happen.

      Thanks to Cyril for the chance to do this..

1ST UPDATE: I (Paul Wratt) am now current (Christmas 2007) lead developer for PHP Navigator.
      After releasing a v4.2 update with fixes and "simple skinning", v4.5 was dropped in
      favour of a v5 complete re-write. I presumed that would be the end of v4.12, however
      due to the nature of the changes and the constant need to prototype functionality as
      well as maintain a wide as possible Web Browser compatibility base, v4.12 continues
      to be the only main "fully working" and non-beta version of PHP Navigator available.
      Once v5 is complete, it will trasition to v4.15 to maintain a historical version,
      which can also be compared on a functional as well as code level with future v5
      builds of PHP Navigator. It is also more usable and compatible than any other
      version available, including the Japanese version which contains FTP access.

==================================
#  PHP Navigator 4.0
#  Author: Cyril Sebastian,
#          Kerala, India.
#  Lead QA: Timothy Alexander
#           Indiana, USA.
#  9:38 PM; May 07, 2007
#  http://navphp.sourceforge.net
==================================

 Web based open source file management system in PHP.
 With Windows XP style & icons. You will feel you are on Windows!

*-* Warning: Provided as is! ABSOLUTELY NO WARRANTY.*-*
*-* The use of this software is at the risk of the user.*-*

For the ORIGINAL version, log on to
'www.sourceforge.net/projects/navphp' or 'navphp.sourceforge.net'

Features
--------
 » User friendly Explorer style navigation and view. Easy to install and configure.
 » Advanced AJAX & DOM to avoid reloading of the full page.
 » gzip & deflate encoding gives you rocket speed data transfer!
 » It inherits the colour scheme of your OS. See the magic of css2!
 » Thumbnail view for wbmp,jpg,png,gif + many others.
 » Right click context menu (may need enabling).
 » Built in HTML editor
 » Keyboard shortcuts
 » Download folder as zip.
 » Extract zip archive to current folder.
 » And the BEST OF ALL, it's FREE!! 
 
Why PHP Navigator?
------------------
  Why yet another web file manager? Because the technology changed, browsers changed,
and file managers too! When the web flies on Web 2.0, AJAX and DOM, why use the same old 
sluggish file manager? Here are some features of old fm's: (i) Written without Javascript
for cross-browser support! (ii) Works on PHP 3.0!! (iii) No cookies!!!
Funny! Isn't it?

Known Issues
------------
 » HTML Editor is browser/OS dependent
 » Code Editor is browser/OS dependent
 » Right Click is browser/OS dependent
 » saved file can be 0 bytes in Chrome (on Linux) when file changes 
   charset type. only happens when copy & paste entire line. 
   Its a bug in Chrome, use history arrow to get edited file, Copy all 
   text, Exit, open the file again Paste, Save.

Requirements
------------
PHP 4.3, gd2 library for thumbnail view.
The web browser should support AJAX for the best functioning. (IE6, Mozilla Firefox etc)
It works best with Firefox 2.x. MSIE does not support some DOM features. It should work
on all other browsers that support standard DOM, else works in normal mode. (No AJAX)

Installing
----------

If you want to install in a remote server, you can use the 'navphp-zip-installer' to
save time & energy. Installer is available from the project website.

---[OR]----

1.Unzip and upload the files into a folder in your site.

2.Edit the 'config.php' to customize to your needs. Set the username and password in
  'config.php' and enable/disable login. (consider using httaccess for password protection,
  if security is an issue. Just look at the documentation for Apache on how to do that.)

3.Edit the 'config_patch.php' to enable 'View in Browser'. Set the FULL PATH to the ROOT 
  of your web server and the corresponding browser path (the one that will get you to the 
  server path). You may need to use PHP Navigator to determine the correct server path. 
  The examples given in 'config_patch.php' are for a local apache server on linux, set to 
  use 'user home' as well as a normal/standard 'wwwroot/htdocs' root.

4.Then access 'index.php' in your browser. If you directly access 'windows.php', proper
  browser detection may NOT OCCUR (choose "Configure/Default/Save" to reset).

Theme: The file named windows.css will be used for theming. Two theme files are provided,
XP & 98. XP is the default; so if you prefer to use 98, rename 'windows.css' and
'windows98.css' accordingly. See "SKINNING.TXT" for more information.

Skins: PHP Navigator v4.12 supports intermediate skinning, ie. "./skins/_name_/skin.css" and a folder
".skins/_name_/*.gif". If no CSS is found, the default one is used, if an icon or image is 
not found, again the default one is used. Choose a default skin in 'config.php', or change it 
through 'Configure PHP Navigator'. See "SKINNING.TXT" for more information.

How to use:
-----------
 Select File	:- 	
			You can select a file by clicking on the icon. [Or Use Arrow keys]
			Details of the selected file will be shown in the left pane.

 Open Folder	:- 	[Enter]
			Double Click a folder. 

 Edit File	:-	[Enter]
			Double click a file for editing. 

 Delete 	:-	[Shift+Ctrl+X]
			Select the file and click delete button.
			You will be prompted for a confirmation

 Rename 	:-	 [F2]
			Select the file/folder and click rename button.
			You will be prompted for the new name.

 Chmode 	:-	Change permissions of files. eg: Php scripts requires 755.

 Copy   	:-	[Shift+Ctrl+C]
			Click on a file icon and click the copy button. 
			You will be prompted for the destination folder.

 Search 	:- 	[Shift+Ctrl+F]
			Click on the 'Search' link on the left, follow the prompts.

 Thumbnail	:- 	[Shift+Ctrl+T]
			Select the file and click the thumbnail button or
			Enable in 'config.php' for all images. Available for jpg, gif, png
			and Windows bitmap.

 Download	:- 	Click on the file name below its icon.

 Extract	:-	[Shift+Ctrl+E]
			Select the zip file and click the extract button. 

 Edit HTML	:- 	[Shift+Ctrl+H]
			Select the HTML file and click the link on the left pane/menu. 

 Edit Code	:- 	[Shift+Ctrl+S]
			Select the source code file and click the link on the left pane/menu. 
			Color coded source code currently supports: html, javascript, php, 
			css, sql, java, text & generic.

 DevEdit	:- 	[Shift+Ctrl+I]
			Select any file and click the link on the pane/menu. 

Trouble-Shooting
----------------
 My picture Thumbnails won't work.
	   Load the gd2 library for PHP. It allows use of thumbnails.
           As of PHP v5, GD2 is installed by default

 How do I password protect PHP Navigator?
	   Read the note at the end of this file.
 
 HTML (WYSIWYG) Editor isn't working in Opera
	   Use IE or Firefox.
 
Permissions on Linux!
	   You need at least 660 permission on working folder for writing. If it was 
	   NOT created or uploaded via PHP, the file or folder may require 770 or 777. 
           You need to "chmod" the them, either manually or using ftp.
 
 My browser doesn't display anything or I just see some erroneous characters.
	   Edit config.php and set '$compress' to false.
           Better still, upgrade your web browser now supports:
              WIN: IE7, FF2; OSX: FF2, Safari3; LINUX: FF2, Opera 9.2, KHTML ?
	
To Do Next
----------
 Languages - create non-english language support
 Modules - archive, media (manipulation & management)
 Multiuser - extend from current v4.32 to support skins (other options)

To Do till v5
-------------
 Skins - integrate other FileWalker skins and modules
 Skin Objects - seperate frontend from backend (easy development)
 Console - "skin" & refactored "backend" functions (standalone)
 New Ajax - To upload files, to get file properties, in parallel.
 Use JSON - standards to support AJAX communication

License:
--------
 Copyright (C) 2006 Cyril Sebastian <cyril2net@yahoo.com>
 Distributed under GNU General Public License

 This script uses zip library by Eric Mueller <eric@themepark.com>, PhpConcept
 Library - Zip Module 2.1 by Vincent Blavet- http://www.phpconcept.net and OEdit
 javascript library by Peter Andreas Harteg - http://www.harteg.dk
 
 Feel free to inform me if you use PHP Navigator in your server.

Known Versions
--------------
Japanese version using ftp for updating
Dutch version 'upload only' for FTP management

Note:
-------
 A basic http authentication is provided with the script. The Login works ONLY IF it is
enabled in 'config.php'. If you already use a password protected folder, you can disable
the user login. For protecting folders the simplest way is basic http authentication
using '.htaccess' files. Refer your host provider for more details.
------------------------------------------------------------------------
Contact: phpnav@isource.net.nz, phpnav-skins@isource.net.nz, devedit@isource.net.nz
Mailing List: (to come)
Web: 'http://phpnav.isource.net.nz/', 'http://skins.phpnav.isource.net.nz'



CHANGES.TXT
OFFICIAL Version 4.1
currently: v4.12.30
dated: 27-02-2018
maintained by: Paul Wratt
-----------------------------------------
 » tested 100% with php5-cgi FastCGI on thttpd & Apache
 » added .tgz .tar.gz .gz and .tar extraction
 » fixed PHP Warnings
 » fixed PHP Strict Warnings
 » fixed PHP inline failure on some servers (old <? to <?php )
 » fixed files: charset=iso-8859-1 => charset=utf-8 (some servers wont recognize them)
 » fixed Upload '$max_size' output
 » changed deprecated 'eregi' to 'stripos'
 » added 'info.php' that can be placed in the webserver root, in place of full "phpinfo.php" or 'server.php'
 » added 'sans-serif' to 'Tahoma' fonts in skin index.html files
 » added missing skin index.html to 'windowsxp' 
 » added known workaround for Basic Authentcation with Appache & php5-cgi FastCGI (needs .htaccess rewrite rule)
 » updated WindowsAlt proto-type
 » updated WindowsXP proto-type
 » updated Details proto-type
 » updated DevPaint/ataripainter.php?dir=path&file=favicon.png
 » added fix for XSS bug in browsers after saving a file
 » re-added missing v4.12.23 fix
 » known bug on some servers: $path_type='root' gives 'real' (needs reworking, again)

Version v4.12.25
-----------------------------------------
 (lost: unknown)

Version v4.12.23
-----------------------------------------
 » fixed AtariST proto-type, desktop window drag in Chrome/Safari

Version v4.12.22
-----------------------------------------
 (lost: unknown)

Version v4.12.21
-----------------------------------------
 (lost: from demo server: maybe 'restrict_to_home' related)

Version v4.12.20
-----------------------------------------
 » fixed tooltip output for folders (remove . and ..)
 » fixed file operations when 'restrict_to_home' is true
 » updated WindowsAlt proto-type
 » added preview.php (only used in demo atm)
 » fixed path when using 'go' (address bar)
 » fixed path when 'restrict_to_home' is true
 » fixed code editor for Chrome & Safari
 » changed browser check for AJAX
 » changed skin setting

Version v4.12.19
-----------------------------------------
 » fixed - minor code cleanup
 » added skin css override (./inc/skin.css)
   (used to develop new css, path compatible with Official PHP Navigator)
 » changed config path defaults
 » added view_charset (best is UTF-8, see: config)
 » fixed security hole when $allow_uploads=false

Version v4.12.18
-----------------------------------------
 » added wget (remote file url to current dir)
 » fixed README.TXT encoding (when using PHP Navigator)

Version v4.12.17
-----------------------------------------
 » added fix for bad linux mono-spaced fonts in Atari proto-type
 » fixed JS error in Atari proto-type (see 'File' ref below)

Version v4.12.16
-----------------------------------------
 » changed edit & save to UTF8
 » added templates into "New File" (see: ./newfiles/*.*)
 » added Mouse Wheel Scrolling support to Atari proto-type
 » added BZip2 compression encoding support (some linux browsers support it for last 12 months)

Version v4.12.15
-----------------------------------------
 » fixed problem with authentication and compression when server automatically adds some html footer types
 » NOTICE: many features have been added to latest Official PHP Navigator v4.40 by Cyril
 » added patch output for servers that add extra data to webserver output

Version v4.12.14
-----------------------------------------
 » fixed upload for Atari Skin
 » updated Atari Skin for Chrome JS error, should fix some other browsers too
 » previous Mozilla update fixes IE7 & IE8 bugs also

Version v4.12.13
-----------------------------------------
 » updated Atari Skin for Mozilla rv:1.9 (FireFox3) (NOTE: DO NOT USE 'File' in the beginning of an (dom) object name/id)
 » fixed minor issues

Version v4.12.12
-----------------------------------------
 » added DevPaint (ataripainter.php) prototype
 » fixed 'browse Here' in  WindowsAlt
 » updated Firefox patches for Iceweasel (freeware Mozilla FF - Debian since Etch (v4)) 
 » updated Firefox patches for BonEcho (freeware Mozilla FF - Damn Small Linux 4.4.8)
 » cleaned up 'invisible' errors (mostly css in Mozilla)
 » standardized code across JS & PHP, functions, pages & includes
 » fixed C:\dir\path translation in address bar (all pathes are now passed as '/')
 » fixed DevEdit for new (moved) functions
 » fixed new left panel folder option links
 » fixed 'download' for Atari ST skin
 » changed 'favicon' for Atari ST skin
 » patched in 'View in Browser' from Atari ST skin
 » can now set default 'startpage' module in config

Version v4.12.11
-----------------------------------------
 » updated compression methods (none/deflate/gzip)
 » updated WindowsAlt proto-type skin
 » fixed IE bugs
 » updated for IE7
 » updated for Safari & OSX
 » added 'DevEdit' proto-type external editor
 » extended Atari ST module (explorer & editor)
 » updated for Windows without proper Terminal font
 » NOTE: pre-WinXPSP3: Terminal (builtin); WinXPSP3, FF: MS Gothic (TTF); Linux, OSX: Andale Mono (TTF);

Version v4.12.10
-----------------------------------------
 » modified javascript include files
 » added 'atari' skin (Atari ST)
 » added 'atari.php' proto-type module (explorer & editor)
 » added 'windowsalt.php' proto-type explorer 'skin'
 » added 'windowsxp.php' proto-type explorer 'skin'

Version v4.12.9
------------------------
 » added favicon for 'search' 
 » added new 'search' dialog
 » fixed saving '\?' in both external editors
 » PNG fix for IE =< v6 (by CSS)
 » fixed minor bugs

Version v4.12.8
------------------------
 » supply both 'lite' & 'full' (lite only has Holocross & Nuvola-details skins)
 » add 'favicon' support
 » changed 'tooltip' timeout to clickable
 » added 'nuvola-details' skin
 » changed 'Configure PHP Navigator' to more flexible skins
 » fixed 'search' to work with details and skins
 » added 'View as Details' to 'File and Folder Tasks'
 » added 'glassplate' full skin
 » add favourites (needs permissions 660,666 or 770,777)

chmod.php
 <?php
#---------------------------
# PHP Navigator 3.2 (4.0)
# dated: 03-8-2006
# Coded by: Cyril Sebastian,
# Kerala,India
# web: navphp.sourceforge.net
#---------------------------
# PHP Navigator 4.12.20
# dated: 20-07-2007
# edited: 07-06-2011
# Modified by: Paul Wratt
# Melbourne, Australia
# Auckland, New Zealand
# web: phpnav.isource.net.nz
#---------------------------


$dir     = @$_REQUEST['dir'];
$file    = @$_REQUEST['file'];
$change  = @$_REQUEST['change'];
$ajax    = isset($_REQUEST['ajax']);
$details = isset($_REQUEST['details']);

$perm = setperms($change);
$file = urldecode($file);
$dir  = urldecode($dir);

include_once("config.php");
include_once("functions.php");

getcookies();

include_once("skin.php");

$reply = 0;

authenticate();							#user login

chdir($dir);

if (!file_exists($file))
  $msg = "Error: '$file' does not exist!";
elseif(chmod($file,$perm)) {
  $msg = "'$file' permissions changed to $change";
  $reply = 1;
}else
  $msg = "Error: Unable to change '$file' permissions to $change!";

function setperms($perm){					# <-- string
  if($perm=="777") return 0777;
  if($perm=="775") return 0775;
  if($perm=="770") return 0770;
  if($perm=="755") return 0755;
  if($perm=="750") return 0755;
  if($perm=="700") return 0700;
  if($perm=="666") return 0666;
  if($perm=="665") return 0665;
  if($perm=="660") return 0660;
  if($perm=="644") return 0644;
  if($perm=="600") return 0600;
  if($perm=="444") return 0444;
  if($perm=="004") return 0444;
  eval("return $perm;");
}

if($ajax){
  expired();
  print"|$reply|$msg|";
  if ($reply){
    if (!$details) filestatus($file)."|";
    if ($details) filedetails($file)."|";
  }
}